targets/core

```diff
@@ -106,17 +106,63 @@ EOF
     apt-get -y dist-upgrade
 fi
 
+# Update the portage tree for Gentoo, if at least one day has passed
+if [ "${DISTROAKA:-"$DISTRO"}" = 'gentoo' ]; then
+    # Get the time of the last sync
+    LAST_SYNC="$(cut -f1 -d' ' /usr/portage/metadata/timestamp.x)"
+    # Add a day to it, in seconds
+    LAST_SYNC="$((LAST_SYNC + 86400))"
+    # Get the current epoch time in seconds
+    TIME_SEC="$(date +%s)"
+    # If at least a day has passed, do the sync
+    if [ "$TIME_SEC" -gt "$LAST_SYNC" ]; then
+        emerge --sync
+    fi
+
+    # Make sure that the system is up to date
+    emerge world -Duv --newuse
+fi
+
 # On release upgrade, keyboard-configuration might be reconfigured.
 fixkeyboardmode
 
+if [ "${DISTROAKA:-"$DISTRO"}" = 'gentoo' ]; then
+    # Install gentoolkit for euse, so we can set use flags
+    # Install cpuid2cpuflags so we can easily set reasonable CFLAGS
+    # NOTE: This tool isn't really supported on ARM yet, update this when it is
+    if [ "$ARCH" = 'x86' ] || [ "$ARCH" = 'amd64' ]; then
+        install gentoolkit cpuid2cpuflags
+        CPUFLAGS="$(cpuinfo2cpuflags-x86 | cut -d: -f2)"
+        sed -i "s|CPU_FLAGS_X86.*|CPU_FLAGS_X86=\"${CPUFLAGS}\"|" \
+            /etc/portage/make.conf
+    fi
+    
+    # Set the dbus flag globally
+    set_flags dbus
+fi
+
+
 # Install critical packages
-install --minimal sudo wget ca-certificates apt-transport-https
+install --minimal gentoo=app-admin/sudo,sudo wget ca-certificates \
+    debian=apt-transport-https,
+
+# Gentoo doesn't have a sudo group, add it now and allow those users to use sudo
+# Debian uses GID 28 for sudo
+if [ "${DISTROAKA:-"$DISTRO"}" = 'gentoo' ]; then
```

Comment:
Do you need this test? Since other distros have a sudo group, the test below will fail and the group won't be created?